### PR TITLE
Add azure_subscription resource

### PR DESCRIPTION
**Pull request by busi-reddy-karnati**
Add `azure_subscription` resource for Azure Subscription onboarding.

### CODE REVIEW

File: api/schema/azure_subscription_resource.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+var (
+	azureSubscriptionResource = Resource{
+		TypeName: "azure_subscription",
+		Schema: resource_schema.Schema{
```
Comment (busi-reddy-karnati) [inline api/schema/azure_subscription_resource.go:18]: Should we also consider management_account_id as an optional parameter?

Comment (rlenglet) [reply api/schema/azure_subscription_resource.go:18]: No, that shouldn't be needed.
The tenant ID is sufficient to group subscriptions together.

File: examples/resources/illumio-cloudsecure_azure_subscription/resource.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (rlenglet) [inline examples/resources/illumio-cloudsecure_azure_subscription/resource.tf:1]:
```suggestion
```

File: api/schema/azure_subscription_resource.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+			Version:     1,
+			Description: "Manages an Azure subscription in CloudSecure.",
+			Attributes: map[string]resource_schema.Attribute{
```
Comment (rlenglet) [inline api/schema/azure_subscription_resource.go:21]: Reorder the attributes to be in alphabetic order (except `id` which should remain on top).

**Review by rlenglet — COMMENTED** [review]
Looks pretty good, thanks!

**Review by rlenglet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**